### ISSUE TITLE
LNURLp issue with Phoenixd backend by bitcarrot

### ISSUE THREAD

**Issue by Funman2** [lnbits/lnbits#2478]
When using the new lnbits integration by bitcarrot https://github.com/bitkarrot/lnbits-phoenixd the Lnurlp extension in lnbits does not work. I guess this is due to phoenixd inability to process Lnurl strings or could it be something else?

Alby for instance opens the pay string, but when trying to pay i get an error message: 
![grafik](https://github.com/lnbits/lnbits/assets/105175715/d618000e-64b3-4911-a5d2-63ae707ff231)

**Comment by DoktorShift** [comment]
I can sign this.

I tested a few extensions with my phoenixd backed lnbits. LNURLp seems to be not working. See here.

https://github.com/bitkarrot/lnbits-phoenixd/issues/1

**Comment by Funman2** [comment]
here are my error messages from container logs:


      

During handling of the above exception, another exception occurred:

Traceback (most recent call last):

  File "/root/.cache/pypoetry/virtualenvs/lnbits-9TtSrW0h-py3.10/lib/python3.10/site-packages/starlette/middleware/errors.py", line 164, in __call__

    await self.app(scope, receive, _send)

  File "/root/.cache/pypoetry/virtualenvs/lnbits-9TtSrW0h-py3.10/lib/python3.10/site-packages/starlette/middleware/base.py", line 191, in __call__

    response = await self.dispatch_func(request, call_next)

  File "/root/.cache/pypoetry/virtualenvs/lnbits-9TtSrW0h-py3.10/lib/python3.10/site-packages/slowapi/middleware.py", line 136, in dispatch

    response = await call_next(request)

  File "/root/.cache/pypoetry/virtualenvs/lnbits-9TtSrW0h-py3.10/lib/python3.10/site-packages/starlette/middleware/base.py", line 165, in call_next

    raise app_exc

  File "/root/.cache/pypoetry/virtualenvs/lnbits-9TtSrW0h-py3.10/lib/python3.10/site-packages/starlette/middleware/base.py", line 151, in coro

    await self.app(scope, receive_or_disconnect, send_no_error)

  File "/root/.cache/pypoetry/virtualenvs/lnbits-9TtSrW0h-py3.10/lib/python3.10/site-packages/starlette/middleware/base.py", line 191, in __call__

    response = await self.dispatch_func(request, call_next)

  File "/app/lnbits/middleware.py", line 215, in block_allow_ip_middleware

    return await call_next(request)

  File "/root/.cache/pypoetry/virtualenvs/lnbits-9TtSrW0h-py3.10/lib/python3.10/site-packages/starlette/middleware/base.py", line 165, in call_next

    raise app_exc

  File "/root/.cache/pypoetry/virtualenvs/lnbits-9TtSrW0h-py3.10/lib/python3.10/site-packages/starlette/middleware/base.py", line 151, in coro

    await self.app(scope, receive_or_disconnect, send_no_error)

  File "/root/.cache/pypoetry/virtualenvs/lnbits-9TtSrW0h-py3.10/lib/python3.10/site-packages/starlette/middleware/base.py", line 191, in __call__

    response = await self.dispatch_func(request, call_next)

  File "/app/lnbits/middleware.py", line 215, in block_allow_ip_middleware

    return await call_next(request)

  File "/root/.cache/pypoetry/virtualenvs/lnbits-9TtSrW0h-py3.10/lib/python3.10/site-packages/starlette/middleware/base.py", line 165, in call_next

    raise app_exc

  File "/root/.cache/pypoetry/virtualenvs/lnbits-9TtSrW0h-py3.10/lib/python3.10/site-packages/starlette/middleware/base.py", line 151, in coro

    await self.app(scope, receive_or_disconnect, send_no_error)

  File "/root/.cache/pypoetry/virtualenvs/lnbits-9TtSrW0h-py3.10/lib/python3.10/site-packages/starlette/middleware/base.py", line 191, in __call__

    response = await self.dispatch_func(request, call_next)

  File "/app/lnbits/middleware.py", line 230, in first_install_middleware

    return await call_next(request)

  File "/root/.cache/pypoetry/virtualenvs/lnbits-9TtSrW0h-py3.10/lib/python3.10/site-packages/starlette/middleware/base.py", line 165, in call_next

    raise app_exc

  File "/root/.cache/pypoetry/virtualenvs/lnbits-9TtSrW0h-py3.10/lib/python3.10/site-packages/starlette/middleware/base.py", line 151, in coro

    await self.app(scope, receive_or_disconnect, send_no_error)

  File "/root/.cache/pypoetry/virtualenvs/lnbits-9TtSrW0h-py3.10/lib/python3.10/site-packages/starlette/middleware/base.py", line 191, in __call__

    response = await self.dispatch_func(request, call_next)

  File "/app/lnbits/middleware.py", line 230, in first_install_middleware

    return await call_next(request)

  File "/root/.cache/pypoetry/virtualenvs/lnbits-9TtSrW0h-py3.10/lib/python3.10/site-packages/starlette/middleware/base.py", line 165, in call_next

    raise app_exc

  File "/root/.cache/pypoetry/virtualenvs/lnbits-9TtSrW0h-py3.10/lib/python3.10/site-packages/starlette/middleware/base.py", line 151, in coro

    await self.app(scope, receive_or_disconnect, send_no_error)

  File "/app/lnbits/middleware.py", line 125, in __call__

    await self.app(scope, receive, send)

  File "/app/lnbits/middleware.py", line 61, in __call__

    await self.app(scope, receive, send)

  File "/root/.cache/pypoetry/virtualenvs/lnbits-9TtSrW0h-py3.10/lib/python3.10/site-packages/starlette/middleware/sessions.py", line 83, in __call__

    await self.app(scope, receive, send_wrapper)

  File "/app/lnbits/middleware.py", line 104, in __call__

    await super().__call__(scope, receive, send)

  File "/root/.cache/pypoetry/virtualenvs/lnbits-9TtSrW0h-py3.10/lib/python3.10/site-packages/starlette/middleware/gzip.py", line 24, in __call__

    await responder(scope, receive, send)

  File "/root/.cache/pypoetry/virtualenvs/lnbits-9TtSrW0h-py3.10/lib/python3.10/site-packages/starlette/middleware/gzip.py", line 44, in __call__

    await self.app(scope, receive, self.send_with_gzip)

  File "/root/.cache/pypoetry/virtualenvs/lnbits-9TtSrW0h-py3.10/lib/python3.10/site-packages/starlette/middleware/cors.py", line 83, in __call__

    await self.app(scope, receive, send)

  File "/root/.cache/pypoetry/virtualenvs/lnbits-9TtSrW0h-py3.10/lib/python3.10/site-packages/starlette/middleware/exceptions.py", line 62, in __call__

    await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)

  File "/root/.cache/pypoetry/virtualenvs/lnbits-9TtSrW0h-py3.10/lib/python3.10/site-packages/starlette/_exception_handler.py", line 64, in wrapped_app

    raise exc

  File "/root/.cache/pypoetry/virtualenvs/lnbits-9TtSrW0h-py3.10/lib/python3.10/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app

    await app(scope, receive, sender)

  File "/root/.cache/pypoetry/virtualenvs/lnbits-9TtSrW0h-py3.10/lib/python3.10/site-packages/starlette/routing.py", line 758, in __call__

    await self.middleware_stack(scope, receive, send)

  File "/root/.cache/pypoetry/virtualenvs/lnbits-9TtSrW0h-py3.10/lib/python3.10/site-packages/starlette/routing.py", line 778, in app

    await route.handle(scope, receive, send)

  File "/root/.cache/pypoetry/virtualenvs/lnbits-9TtSrW0h-py3.10/lib/python3.10/site-packages/starlette/routing.py", line 299, in handle

    await self.app(scope, receive, send)

  File "/root/.cache/pypoetry/virtualenvs/lnbits-9TtSrW0h-py3.10/lib/python3.10/site-packages/starlette/routing.py", line 79, in app

    await wrap_app_handling_exceptions(app, request)(scope, receive, send)

  File "/root/.cache/pypoetry/virtualenvs/lnbits-9TtSrW0h-py3.10/lib/python3.10/site-packages/starlette/_exception_handler.py", line 64, in wrapped_app

    raise exc

  File "/root/.cache/pypoetry/virtualenvs/lnbits-9TtSrW0h-py3.10/lib/python3.10/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app

    await app(scope, receive, sender)

  File "/root/.cache/pypoetry/virtualenvs/lnbits-9TtSrW0h-py3.10/lib/python3.10/site-packages/starlette/routing.py", line 74, in app

    response = await func(request)

  File "/root/.cache/pypoetry/virtualenvs/lnbits-9TtSrW0h-py3.10/lib/python3.10/site-packages/fastapi/routing.py", line 299, in app

    raise e

  File "/root/.cache/pypoetry/virtualenvs/lnbits-9TtSrW0h-py3.10/lib/python3.10/site-packages/fastapi/routing.py", line 294, in app

    raw_response = await run_endpoint_function(

  File "/root/.cache/pypoetry/virtualenvs/lnbits-9TtSrW0h-py3.10/lib/python3.10/site-packages/fastapi/routing.py", line 191, in run_endpoint_function

    return await dependant.call(**values)

  File "/app/lnbits/extensions/lnurlp/lnurl.py", line 91, in api_lnurl_callback

    payment_hash, payment_request = await create_invoice(

  File "/app/lnbits/core/services.py", line 154, in create_invoice

    ) = await funding_source.create_invoice(

  File "/app/lnbits/wallets/phoenixd.py", line 91, in create_invoice

    raise UnsupportedError("description_hash")

lnbits.wallets.base.UnsupportedError: description_hash

2024-05-08 07:34:04.81 | ERROR | Exception: description_hash

2024-05-08 07:34:04.81 | INFO | 31.19.129.130:0 - "GET /lnurlp/api/v1/lnurl/cb/nzdP83?amount=1000&comment=test HTTP/1.1" 500

2024-05-08 07:34:04.81 | ERROR | Exception in ASGI application

2024-05-08 07:34:06.16 | INFO | 31.19.129.130:0 - "GET /lnurlp/api/v1/links?all_wallets=true HTTP/1.1" 200

2024-05-08 07:34:26.12 | INFO | 31.19.129.130:0 - "GET /lnurlp/api/v1/links?all_wallets=true HTTP/1.1" 200

**Comment by Bashy** [comment]
in `lnbits/wallets/phoenixd.py`
```py
        if description_hash or unhashed_description:
            raise UnsupportedError("description_hash")
```

Commenting those 2 lines seems to solve the problem, but this might need further tests.

**Comment by bitkarrot** [comment]
https://github.com/ACINQ/phoenixd/pull/47

**Comment by bitkarrot** [comment]
https://github.com/ACINQ/phoenixd/pull/50

**Comment by Funman2** [comment]
Great to see progress is beeing made!

**Comment by Funman2** [comment]
https://github.com/bitkarrot/lnbits-phoenixd/issues/1#issuecomment-2115853854

**Comment by Funman2** [comment]
Can solve the issue when replacing phoenix.py in lnbits/wallets with this file:

https://github.com/lnbits/lnbits/blob/11ba5955792fc9dce71387f67c6d9ae45d1d2832/lnbits/wallets/phoenixd.py

**Comment by motorina0** [comment]
@bitkarrot can you confirm this was solved (and close the issue if so)

**Comment by prusnak** [comment]
Should be fixed via 83b89851a59b617970c8f3dc5bef2ade450cb877

Please reopen if it is not

**Comment by bitkarrot** [comment]
@prusnak  just saw your comment. 

Was resolved a while back. there is also a new release of phoenixd out, will need to test this.
